django/contrib/admin/options.py

```diff
@@ -1027,9 +1027,20 @@ def construct_search(field_name):
             orm_lookups = [construct_search(str(search_field))
                            for search_field in search_fields]
             for bit in search_term.split():
-                or_queries = [models.Q(**{orm_lookup: bit})
-                              for orm_lookup in orm_lookups]
-                queryset = queryset.filter(reduce(operator.or_, or_queries))
+                or_queries = []
+                for orm_lookup in orm_lookups:
+                    try:
+                        query = models.Q(**{orm_lookup: bit})
+                        queryset.filter(query)
+                        or_queries.append(query)
+                    except ValueError:
+                        # This piece of the search term failed for this lookup, skip.
+                        pass
```

Comment:
Not sure where my initial comment went but you'll want to only catch the `ValueError` on `queryset.filter()` here. This could silence legitimate `ValueError` on `Q.__init__`.

```python
or_queries = []
for orm_lookup in orm_lookups:
    query = models.Q(**{orm_lookup: bit})
    try:
        queryset.filter(query)
    except ValueError:
        continue
    or_queries.append(query)

Reply:
Makes sense :+1: